conan/internal/cache/cache.py

```diff
@@ -111,10 +111,8 @@ def create_build_pkg_layout(self, pref: PkgReference):
     def recipe_layout(self, ref: RecipeReference):
         """ the revision must exists, the folder must exist
         """
-        if ref.revision is None:  # Latest one
-            ref_data = self._db.get_latest_recipe(ref)
-        else:
-            ref_data = self._db.get_recipe(ref)
+        assert ref.revision is not None
```

Comment:
This was the main change, the docstring said the revision must exists, but then it was checking if not and resolving it.